src/many-ledger/src/storage/ledger_mintburn.rs

```diff
@@ -0,0 +1,105 @@
+use crate::error;
+use crate::storage::ledger_tokens::key_for_symbol;
+use crate::storage::{key_for_account_balance, LedgerStorage};
+use many_error::ManyError;
+use many_modules::ledger::TokenInfoArgs;
+use many_types::ledger::{LedgerTokensAddressMap, Symbol, TokenAmount, TokenInfoSupply};
+use merk::{BatchEntry, Op};
+use std::collections::BTreeSet;
+
+impl LedgerStorage {
+    pub(crate) fn get_token_supply(&self, symbol: &Symbol) -> Result<TokenInfoSupply, ManyError> {
+        Ok(self
+            .info_token(TokenInfoArgs {
+                symbol: *symbol,
+                extended_info: None,
+            })?
+            .info
+            .supply)
+    }
+
+    pub fn mint_token(
+        &mut self,
+        symbol: Symbol,
+        distribution: &LedgerTokensAddressMap,
+    ) -> Result<(), ManyError> {
+        let mut batch: Vec<BatchEntry> = Vec::new();
+        let mut circulating = TokenAmount::zero();
+        for (address, amount) in distribution.iter() {
+            circulating += amount;
+            let new_balance = self
+                .get_multiple_balances(address, &BTreeSet::from([symbol]))?
+                .get(&symbol)
+                .map_or(amount.clone(), |b| b + amount);
+            let key = key_for_account_balance(address, &symbol);
+            batch.push((key, Op::Put(new_balance.to_vec())));
+        }
+
+        // Update circulating supply
+        let mut info = self
+            .info_token(TokenInfoArgs {
+                symbol,
+                extended_info: None,
+            })?
+            .info;
+        info.supply.circulating += circulating;
+
+        // Update the supply total if the circulating supply is greater than the current total
```

Comment:
Right now these two numbers should always be the same. But really the total will always be updated at the same _rate_ as the circulating, so they should move in lock steps.

So when you update circulating please update total at the same time.

Reply:
What about 
```
; ...but if there was a staking
; mechanism, that would be less, as those tokens are not burnt but
; frozen.
```
?

Reply:
Let's presume a pre-condition where circulating = 1000 and total = 1500 (with 500 _somewhere_). Minting (or burning the other way) 200 would lead to circulating being 1200, and total being 1700. With this code it would be 1200/1500. Does that make more sense?